src/docfx/config/Config.cs

```diff
@@ -168,81 +168,62 @@ public IEnumerable<string> GetFileReferences()
-            if (File.Exists(configPath))
-            {
-                return true;
-            }
-
-            configFile = "docfx.json";
-            configPath = PathUtility.NormalizeFile(Path.Combine(parentPath, configFile));
-            if (File.Exists(configPath))
-            {
-                return true;
-            }
-            return false;
-        }
+        /// <summary>
+        /// Load the config if it exists under <paramref name="docsetPath"/> or return default config
+        /// </summary>
+        public static (List<Error> errors, Config config) TryLoad(string docsetPath, CommandLineOptions options, bool extend = true)
+            => LoadCore(docsetPath, options, extend);
 
-        private static (List<Error>, Config) LoadCore(string docsetPath, string configPath, CommandLineOptions options, bool extend)
+        private static (List<Error>, Config) LoadCore(string docsetPath, CommandLineOptions options, bool extend)
         {
-            // Options should be converted to config and overwrite the config parsed from docfx.yml/docfx.json
             var errors = new List<Error>();
             Config config = null;
 
-            var (loadErrors, configObject) = LoadConfigObject(configPath, configPath);
+            var configPath = PathUtility.FindYamlOrJson(Path.Combine(docsetPath, "docfx"));
+            var (loadErrors, configObject) = configPath == null ? (errors, new JObject()) : LoadConfigObject(configPath, configPath);
+
+            // apply options
             var optionConfigObject = Expand(options?.ToJObject());
             var finalConfigObject = JsonUtility.Merge(configObject, optionConfigObject);
+            errors.AddRange(loadErrors);
 
+            // apply global config
             var globalErrors = new List<Error>();
             (globalErrors, finalConfigObject) = ApplyGlobalConfig(finalConfigObject);
             errors.AddRange(globalErrors);
 
+            // apply extends
             if (extend)
             {
                 var extendErrors = new List<Error>();
                 (extendErrors, finalConfigObject) = ExtendConfigs(finalConfigObject, docsetPath);
                 errors.AddRange(extendErrors);
             }
 
+            // apply overwrite
             finalConfigObject = OverwriteConfig(finalConfigObject, options.Locale, GetBranch());
 
             var deserializeErrors = new List<Error>();
             (deserializeErrors, config) = JsonUtility.ToObjectWithSchemaValidation<Config>(finalConfigObject);
             errors.AddRange(deserializeErrors);
 
+            // validate metadata
             errors.AddRange(MetadataValidator.Validate(config.GlobalMetadata, nameof(GlobalMetadata)));
             errors.AddRange(MetadataValidator.Validate(config.FileMetadata, nameof(FileMetadata)));
 
+            config.ConfigFileName = configPath == null
+                ? config.ConfigFileName
```

Comment:
>ConfigFileName [](start = 25, length = 14)

when configPath is null, config.ConfigFileName should be null or empty, instead of using default value `docfx.yml` that does not exist

Reply:
it's just used to fill the error messages, when the config is empty, there should be no errors for configurations

Reply:
add @live1206  for double confirm

Reply:
double checked, yes